tests/common.go

```diff
@@ -6423,3 +6424,134 @@ func DoRetryWithTimeoutWithGinkgoRecover(taskFunc func() (interface{}, bool, err
 	}
 	return task.DoRetryWithTimeout(taskFuncWithGinkgoRecover, timeout, timeBeforeRetry)
 }
+
+
+// GetVolumesInDegradedState Get the list of volumes in degraded state
+func GetVolumesInDegradedState(contexts []*scheduler.Context) ([]*volume.Volume, error) {
+	var volumes []*volume.Volume
+	err := RefreshDriverEndPoints()
+	if err != nil {
+		return nil, err
+	}
+	for _, ctx := range contexts {
+		vols, err := Inst().S.GetVolumes(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, vol := range vols {
+			appVol, err := Inst().V.InspectVolume(vol.ID)
+			if err != nil {
+				return nil, err
+			}
+			log.InfoD(fmt.Sprintf("Current Status of the volume [%v] is [%v]", vol.Name, appVol.Status))
+			if fmt.Sprintf("[%v]", appVol.Status) != "VOLUME_STATUS_DEGRADED" {
+				volumes = append(volumes, vol)
+			}
+		}
+	}
+	return volumes, nil
+}
+
+// VerifyVolumeStatusOnline returns true is volume status is up
+func VerifyVolumeStatusOnline(vol *volume.Volume) error {
+	appVol, err := Inst().V.InspectVolume(vol.ID)
+	if err != nil {
+		return err
+	}
+	if fmt.Sprintf("%v", appVol.Status) != "VOLUME_STATUS_UP" {
+		return fmt.Errorf("volume [%v] status is not up. Current status is [%v]", vol.Name, appVol.Status)
+	}
+	return nil
+}
+
+type kvdbNode struct {
+	PeerUrls   []string `json:"PeerUrls"`
+	ClientUrls []string `json:"ClientUrls"`
+	Leader     bool     `json:"Leader"`
+	DbSize     int      `json:"DbSize"`
+	IsHealthy  bool     `json:"IsHealthy"`
+	ID         string   `json:"ID"`
+}
+
+type KvdbNodes []map[string]kvdbNode
+
+// GetAllKvdbNodes returns list of all kvdb nodes present in the cluster
+func GetAllKvdbNodes() ([]kvdbNode, error) {
+	storageNodes := node.GetStorageNodes()
+	randomIndex := rand.Intn(len(storageNodes))
+	randomNode := storageNodes[randomIndex]
+
+	jsonConvert := func(jsonString string) ([]kvdbNode, error) {
+		var nodes KvdbNodes
+		var kvdb kvdbNode
+		var kvdbNodes []kvdbNode
+
+		err := json.Unmarshal([]byte(fmt.Sprintf("[%s]", jsonString)), &nodes)
+		if err != nil {
+			return nil, err
+		}
+		for _, nodeMap := range nodes {
+			for id, value := range nodeMap {
+				kvdb.ID = id
+				kvdb.Leader = value.Leader
```

Comment:
no need to explicitly convert, jsin unmarshal will take care of it if &kvdbnodes are given

Reply:
fixed